WooCommerce/Classes/POS/Presentation/Barcode Scanning/GameControllerBarcodeParser.swift

```diff
@@ -0,0 +1,197 @@
+        switch keyCode {
+        // Numbers and their shifted symbols
+        case .zero: return isShiftPressed ? ")" : "0"
+        case .one: return isShiftPressed ? "!" : "1"
+        case .two: return isShiftPressed ? "@" : "2"
+        case .three: return isShiftPressed ? "#" : "3"
+        case .four: return isShiftPressed ? "$" : "4"
+        case .five: return isShiftPressed ? "%" : "5"
+        case .six: return isShiftPressed ? "^" : "6"
+        case .seven: return isShiftPressed ? "&" : "7"
+        case .eight: return isShiftPressed ? "*" : "8"
+        case .nine: return isShiftPressed ? "(" : "9"
+
+        // Letters - lowercase without shift, uppercase with shift
+        case .keyA: return isShiftPressed ? "A" : "a"
+        case .keyB: return isShiftPressed ? "B" : "b"
+        case .keyC: return isShiftPressed ? "C" : "c"
+        case .keyD: return isShiftPressed ? "D" : "d"
+        case .keyE: return isShiftPressed ? "E" : "e"
+        case .keyF: return isShiftPressed ? "F" : "f"
+        case .keyG: return isShiftPressed ? "G" : "g"
+        case .keyH: return isShiftPressed ? "H" : "h"
+        case .keyI: return isShiftPressed ? "I" : "i"
+        case .keyJ: return isShiftPressed ? "J" : "j"
+        case .keyK: return isShiftPressed ? "K" : "k"
+        case .keyL: return isShiftPressed ? "L" : "l"
+        case .keyM: return isShiftPressed ? "M" : "m"
+        case .keyN: return isShiftPressed ? "N" : "n"
+        case .keyO: return isShiftPressed ? "O" : "o"
+        case .keyP: return isShiftPressed ? "P" : "p"
+        case .keyQ: return isShiftPressed ? "Q" : "q"
+        case .keyR: return isShiftPressed ? "R" : "r"
+        case .keyS: return isShiftPressed ? "S" : "s"
+        case .keyT: return isShiftPressed ? "T" : "t"
+        case .keyU: return isShiftPressed ? "U" : "u"
+        case .keyV: return isShiftPressed ? "V" : "v"
+        case .keyW: return isShiftPressed ? "W" : "w"
+        case .keyX: return isShiftPressed ? "X" : "x"
+        case .keyY: return isShiftPressed ? "Y" : "y"
+        case .keyZ: return isShiftPressed ? "Z" : "z"
+
+        // Punctuation and symbols with shift variants
+        case .spacebar: return " "
+        case .hyphen: return isShiftPressed ? "_" : "-"
+        case .equalSign: return isShiftPressed ? "+" : "="
+        case .openBracket: return isShiftPressed ? "{" : "["
+        case .closeBracket: return isShiftPressed ? "}" : "]"
+        case .backslash: return isShiftPressed ? "|" : "\\"
+        case .semicolon: return isShiftPressed ? ":" : ";"
+        case .quote: return isShiftPressed ? "\"" : "'"
+        case .comma: return isShiftPressed ? "<" : ","
+        case .period: return isShiftPressed ? ">" : "."
+        case .slash: return isShiftPressed ? "?" : "/"
+        case .graveAccentAndTilde: return isShiftPressed ? "~" : "`"
+        case .returnOrEnter: return "\r"
```

Comment:
Should `\n` be in this list?

Reply:
Since it's `returnOrEnter`, I think they are both treated the same way with `GameController` framework.